acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/AddValidatorsAcceptanceTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.test.acceptance;
+
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.test.acceptance.dsl.AcceptanceTestBase;
+import tech.pegasys.teku.test.acceptance.dsl.BesuNode;
+import tech.pegasys.teku.test.acceptance.dsl.TekuNode;
+import tech.pegasys.teku.test.acceptance.dsl.tools.deposits.ValidatorKeystores;
+
+public class AddValidatorsAcceptanceTest extends AcceptanceTestBase {
+
+  @Test
+  void shouldLoadAdditionalValidatorsWithoutRestart() throws Exception {
+    final BesuNode eth1Node = createBesuNode();
+    eth1Node.start();
+
+    final ValidatorKeystores initialKeystores =
+        createTekuDepositSender().sendValidatorDeposits(eth1Node, 2);
+
+    final ValidatorKeystores additionalKeystores =
+        createTekuDepositSender().sendValidatorDeposits(eth1Node, 2);
+
+    final TekuNode node =
+        createTekuNode(
+            config ->
+                config
+                    .withNetwork("less-swift")
+                    .withDepositsFrom(eth1Node)
+                    .withValidatorKeystores(initialKeystores));
+    node.start();
+
+    node.waitForOwnedValidatorCount(2);
+    node.waitForGenesis();
+
+    node.addValidators(additionalKeystores);
+    node.waitForOwnedValidatorCount(4);
+
+    // If the added validators perform their duties properly, the network will finalize.
+    node.waitForNewFinalization();
+  }
```

Comment:
might be worth sending a second HUP after its finalized once, and check it finalizes again and doesn't die in some weird way... 
2 reasons i guess, 
1> HUP without an actual change
2> second HUP handled successfully and finalizes after...

My guess is people won't do this just once, it'll be part of a workflow, but I'm not sure its worth loading a third set of keys...

Reply:
hmm, waiting for another finalisation adds a fair bit of time.  My main aim with the AT is really just checking that the `HUP` signal really does trigger loading validators - can't really unit test it because it isn't necessarily supported by all JVMs (fortunately ATs use docker).

But I've extended it to one more validators which won't yet be active and send a second HUP to load them so we know that works.

Reply:
sounds like a good compromise, and it still shows multiple HUP.